engine/classes/Elgg/Database/Plugins.php

```diff
@@ -1165,11 +1167,11 @@ public function setPriority(ElggPlugin $plugin, $priority) {
 
 		$name = $this->namespacePrivateSetting('internal', 'priority');
 
-		if (!$plugin->guid) {
+		if (!$plugin->setPrivateSetting($name, $priority)) {
```

Comment:
why did we need to swap this?

Reply:
ElggPlugin::save() sets the priority if there isn't one before the parent::save() is called